providers/implementations/rands/drbg.c

```diff
@@ -658,9 +658,6 @@ int ossl_prov_drbg_generate(PROV_DRBG *drbg, unsigned char *out, size_t outlen,
             || now - drbg->reseed_time >= drbg->reseed_time_interval)
             reseed_required = 1;
     }
-    if (drbg->parent != NULL
-            && get_parent_reseed_count(drbg) != drbg->parent_reseed_counter)
-        reseed_required = 1;
```

Comment:
It is not sufficient to remove these three lines only. There was a lot of fuzz in the past about potential data races related to the reseed counter, which resulted in some tricky tsan fixes, e.g. in commit a83dc59afa2e0207180d7218efed19b20d48de95  by @bernd-edlinger (see #7399, #7394).

If the downward propagation of reseedings is not wanted anyore then the entire reseed_counter implemention is obsolete and needs to be removed, including all the magic tsan fixes.

Reply:
Good point.  I am loathe to make such changes if this is going to 3.0.  For master, there is no problem doing so.

Reply:
A compromise would be to backport only your three-line fix and do the ripping-out in master only. I heared you have some practice with bulldozers now, that should be very helpful.

Reply:
OTOH: these three lines *are the breaking* change, the tsan fixes are just technical gimmicks compared to it.

Reply:
Removing all the code is actually easy: just remove the `reseed_counter` and `parent_reseed_counter` members from the struct and follow the compiler errors ;-)

Reply:
Perhaps we should not make the change but only increment the reseed counter if it comes via RAND_seed()?
That wouldn't be a breaking change.

Reply:
Removing the code from the DRBG is easy.  Making the tests useful after doing so isn't so easy.

Reply:
I've done the removal as a separate commit.  See what you think.
The changes aren't too large.

Reply:
I don’t know when I will be able to take a look. I have a busy week in the office.

Reply:
> ~Perhaps we should not make the change but only increment the reseed counter if it comes via RAND_seed()? That wouldn't be a breaking change.~
> 
> Doesn't work.

In which way?

Reply:
It required libcrypto incrementing values inside the provider which isn't good.
I don't think a new provider API is justified.

Now, it might be possible to handle the reseeding inside the EVP code.  I didn't dive into this option.